### DISCUSSION TITLE
Rewrite the parser without global variables

### DISCUSSION THREAD

**Discussion by awerqo** [projectdiscovery/katana#1050] [Ideas]
When we use katana as a library and initialize the parser, it turns out to be one for the entire project (since it stores the settings in a global variable). Moreover, there is no function to reset these settings.

Maybe we should abandon global variables and put them in a structure so that we can use different site parsing parameters for different targets within one program?

**Comment by awerqo** [comment]
Has anyone figured out how to use this in library mode with different parser parameters for different targets? It seems completely impossible (due to parser options global for everything)

https://github.com/projectdiscovery/katana/blob/main/pkg/engine/parser/parser.go#L36
https://github.com/projectdiscovery/katana/blob/main/pkg/engine/parser/parser_generic.go#L16

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
